File: Python/141A. Amusing Joke.py
def main():
    s = list(input())
    t = list(input())
    a = s + t
    vecsize = len(a)
    u = list(input())
    for i in range(len(u)):
        for l in range(len(a)):
            if u[i] == a[l]:
                a.pop(l)
                break
    if len(a)==0 and vecsize==len(u):
        print("YES\n")
    else:
        print("NO\n")
main()

File: C++/141A. Amusing Joke.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    string s, t, u;
    cin>>s;
    vector<char> a;
    for(int i=0; i<s.size(); i++){
        a.push_back(s[i]);
    }
    cin>>t;
    for(int i=0; i<t.size(); i++){
        a.push_back(t[i]);
    }
    cin>>u;
    int vecsize = a.size();
    for(int i=0; i<u.size(); i++){
        for(int l=0; l<a.size(); l++){
            if(u[i]==a[l]){
                a.erase(a.begin()+l);
                break;
            }
        }
    }
    if(a.size()==0&&vecsize==u.size()){
        cout<<"YES"<<endl;
    }
    else{
        cout<<"NO"<<endl;
    }
    return 0;
}